test/e2e/framework.go

```diff
@@ -293,6 +296,84 @@ func (data *TestData) createNamespace(namespace string, mutators ...func(ns *cor
 	return nil
 }
 
+// createVCNamespace creates a VC namespace with the provided namespace.
+func (data *TestData) createVCNamespace(namespace string) error {
+	err := testData.vcClient.startSession()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		testData.vcClient.closeSession()
+	}()
+
+	svID, _ := data.vcClient.getSupervisorID()
+	vcNamespace := &VCNamespaceCreateSpec{
+		Supervisor: svID,
+		Namespace:  namespace,
+		NetworkSpec: InstancesNetworkConfigInfo{
+			NetworkProvider: "NSX_VPC",
+			VpcNetwork: InstancesVpcNetworkInfo{
+				DefaultSubnetSize: 16,
+			},
+		},
+	}
+	dataJson, err := json.Marshal(vcNamespace)
+	if err != nil {
+		log.Error(err, "Unable convert vcNamespace object to json bytes", "namespace", namespace)
+		return fmt.Errorf("unable convert vcNamespace object to json bytes: %v", err)
+	}
+	request, err := data.vcClient.prepareRequest(http.MethodPost, createVCNamespaceEndpoint, dataJson)
+	if err != nil {
+		log.Error(err, "Failed to prepare http request with vcNamespace data", "namespace", namespace)
+		return fmt.Errorf("failed to parepare http request with vcNamespace data: %v", err)
+	}
+	if _, err = data.vcClient.handleRequest(request, nil); err != nil {
+		log.Error(err, "Failed to create VC namespace", "namespace", namespace)
+		return err
+	}
+	// wait for the namespace on k8s running
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, defaultTimeout, false, func(ctx context.Context) (done bool, err error) {
+		ns, err := data.clientset.CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+		if err != nil {
+			log.Error(err, "Check namespace existence", "namespace", namespace)
+			return false, err
+		}
+		if ns.Status.Phase == corev1.NamespaceActive {
```

Comment:
A K8s Namespace's phase is already Active after it is created and before it is deleted. I think you check the Namespace's condition type "NamespaceNetworkReady" with status "True" which is introduced in nsx-operator to identify if the VPC is realized?

Reply:
ok